backend/src/main/java/net/vidux/camhub/discovery/RawCameraDataFactory.java

```diff
@@ -0,0 +1,28 @@
+package net.vidux.camhub.discovery;
+
+class RawCameraDataFactory {
+  private static final String regex = "\\t";
+
+  String extractSerialNumber(String productNumber) {
+    int endIndex = productNumber.length() - 1;
+    final char[] chars = productNumber.toCharArray();
+    while (chars[endIndex] > '9' || chars[endIndex] < '0') {
```

Comment:
Ha az expression egy kicsit is bonyolult, nevesítsük meg függvénnyel, hogy  a kód olvasónak egyszerűbb legyen a dolga.